File: back-end/src/main/java/org/example/entities/Section.java
package org.example.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data @AllArgsConstructor @NoArgsConstructor
@Entity
@Table(name="sections")
public class Section {

    @Id
    @GeneratedValue(strategy = GenerationType.TABLE)
    long id;

    @Column
    String codeSection;

    @Column
    String codeSpecialite;

    @Column
    String nomSection;

    @Column
    int nombreGroup;

    @Column
    int capaciteApprenant;

    @Column
    String codeDiplome;

    @Column
    String dateEntree;

    @Column
    String dateFin;

    @OneToMany(fetch = FetchType.LAZY)
    List<Apprenant> listApprenant;

    @ManyToOne(fetch = FetchType.LAZY)
    SessionEntity session;
}

File: front-end/src/app/models/Section.ts
import { Session } from "./Session";


export class Section {
    
	id : number;
	codeSection  :  string;
	codeSpecialite  : string ;
    nomSection : string;
	nombreGroup  : Number;
	capaciteApprenant : Number;
	codeDiplome : string;
	dateEntree : string;
	dateFin : string;
    session? : Session

    constructor(
        id : number,
    codeSection : string,
    codeSpecialite  : string ,
    nomSection : string,
	nombreGroup  : Number,
	capaciteApprenant : Number,
	codeDiplome : string,
	dateEntree : string,
	dateFin : string,
    session? : Session
    ){
        this.id = id
        this.codeSection = codeSection;
        this.codeSpecialite = codeSpecialite;
        this.nomSection = nomSection;
        this.nombreGroup = nombreGroup;
        this.capaciteApprenant = capaciteApprenant;
        this.codeDiplome = codeDiplome;
        this.dateEntree = dateEntree;
        this.dateFin = dateFin;
        this.session = session;
    }

    
}